Carta.py

```diff
@@ -320,33 +341,73 @@ def opponentRespondsInGrabPhase(self):
             else:
                 print("Carta.py: opponentRespondsInGrabPhase: opponent does not touch any card")
 
+    def removeCurrentGrabbingCard(self):
+        for i in range(len(self.grabbingCardsInPlay)):
+            if (self.grabbingCardsInPlay[i].getLastWord() == self.curReadingCard.getLastWord()):
+                self.grabbingCardsInPlay[i].setFrame(None)
+                self.grabbingCardsInPlay[i].setStatus(GrabCardStatus.INVALID)
+                self.usedGrabbingCards.cards.append(self.grabbingCardsInPlay.pop(i))
+
+    def shouldOpponentTransfer(self, yourFalseTouch):
+        if (self.GPinfo.oppoGrabCardSide == GrabbingCardSide.YOUR_SIDE):
+            self.numYourGrabCardInPlay -= 1
+            self.phase = Phase.OPPONENT_ADJUSTMENT
+            self.numCardsToTransfer += 1
+        elif (self.GPinfo.oppoGrabCardSide == GrabbingCardSide.OPPONENT_SIDE):
+            self.numOppoGrabCardInPlay -= 1
+
+        if (yourFalseTouch):
+            self.numCardsToTransfer += 1
+            self.phase = Phase.OPPONENT_ADJUSTMENT
+
     def decideWinner(self):
         statement = ""
+        self.findPlayersGrabCardSide()
         # Both players grabbed the right card, but you grabbed it faster than opponent
         if ((self.GPinfo.yourGrabCardLastWord == self.curReadingCard.getLastWord()) and \
             (self.GPinfo.oppoGrabCardLastWord == self.curReadingCard.getLastWord()) and \
             (self.GPinfo.yourTime < self.GPinfo.opponentTime)):
             self.GPinfo.youWin = True
             self.GPinfo.opponentWin = False
             statement = "Carta.py: decideWinner: Both players grabbed the right card, but you grabbed it faster than opponent"
+            self.dialogs.append("You Win, Opponent Loses")
+            if (self.GPinfo.yourGrabCardSide == GrabbingCardSide.YOUR_SIDE):
+                self.numYourGrabCardInPlay -= 1
+            elif (self.GPinfo.yourGrabCardSide == GrabbingCardSide.OPPONENT_SIDE):
+                self.numOppoGrabCardInPlay -= 1
+                self.phase = Phase.YOUR_ADJUSTMENT
+                self.numCardsToTransfer += 1
         # Both players grabbed the right card, but opponent grabbed it faster than you
         elif ((self.GPinfo.yourGrabCardLastWord == self.curReadingCard.getLastWord()) and \
               (self.GPinfo.oppoGrabCardLastWord == self.curReadingCard.getLastWord()) and \
               (self.GPinfo.yourTime > self.GPinfo.opponentTime)):
             self.GPinfo.youWin = False
             self.GPinfo.opponentWin = True
             statement = "Carta.py: decideWinner: Both players grabbed the right card, but opponent grabbed it faster than you"
-        # you didn't grab the right card, but opponent grabbed the right card
+            self.dialogs.append("Opponent Wins, You Lose")
```

Comment:
For example, move this out.

if (you win and opponent win)
      append("You win opponent win")
elif ...

elif ..

else

Reply:
```
def appendDialogsWhenDecidingWinner:
      # append win lose info
      # append who needs to adjust
```